wls-broadcast-service/src/test/java/de/muenchen/oss/wahllokalsystem/rest/BroadcastControllerTest.java

```diff
@@ -0,0 +1,171 @@
+package de.muenchen.oss.wahllokalsystem.rest;
+
+import static de.muenchen.oss.wahllokalsystem.TestConstants.SPRING_NO_SECURITY_PROFILE;
+import static de.muenchen.oss.wahllokalsystem.TestConstants.SPRING_TEST_PROFILE;
+import de.muenchen.oss.wahllokalsystem.domain.Message;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.FachlicheWlsException;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.util.ExceptionKonstanten;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import de.muenchen.oss.wahllokalsystem.MicroServiceApplication;
+import de.muenchen.oss.wahllokalsystem.domain.MessageRepository;
+import de.muenchen.oss.wahllokalsystem.service.BroadcastService;
+import de.muenchen.oss.wahllokalsystem.utils.BroadcastSecurityUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest(
+        classes = { MicroServiceApplication.class },
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        properties = {
+                "spring.datasource.url=jdbc:h2:mem:wahllokalsystem;DB_CLOSE_ON_EXIT=FALSE",
+                "refarch.gracefulshutdown.pre-wait-seconds=0"
+        }
+)
+@ActiveProfiles(profiles = { SPRING_TEST_PROFILE, SPRING_NO_SECURITY_PROFILE })
+class BroadcastControllerTest {
```

Comment:
Der Name passt nicht zu den Tests. Die Test laufen gegen den Service.

Für den Controller reicht aus meiner Sicht ein Unit-Tests mit Mockito.

Das Gesamtzusammenspiel wird über die `BroadcastIntegrationsTests` verifiziert.

Die Tests die hier sind bitte als Unit Tests für `BroadcastService` umbauen. Für den Unittest brauchen wir den Springcontext nicht.

Reply:
resolved in 2412ef4d01b26f3637e32171dbf9b5a19ee6c39a